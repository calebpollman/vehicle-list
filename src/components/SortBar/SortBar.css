.sort-bar-container {
  align-items: center;
  background-color: var(--color-sort-bar-background);
  display: flex;
  height: 4%;
  justify-content: flex-end;
  width: 100%;
}

.sort-bar-text {
  color: var(--color-sort-bar-text);
  font-family: var(--font-application);
  font-size: var(--font-size-small);
  margin-right: 2%;
}