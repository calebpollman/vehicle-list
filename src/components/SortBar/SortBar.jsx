import React from 'react';
import PropTypes from 'prop-types';
import SortButton from '../SortButton/SortButton';
import './SortBar.css';

const SortBar = ({sortSelection, sortOptions, updateSortSelection}) => {
  sortOptions = sortOptions.map(sortOption => {
    return (
      <SortButton 
        key={sortOption}
        isActive={sortOption.toLowerCase() === sortSelection ? true : false}
        sortOption={sortOption}
        updateSortSelection={updateSortSelection}
      />
    );
  });

  return (
    <div className="sort-bar-container">
      <label className="sort-bar-text">Sort Vehicles By:</label>
      {sortOptions}
    </div>
  );
}

SortBar.propTypes = {
  sortSelection: PropTypes.string.isRequired,
  sortOptions: PropTypes.array.isRequired,
  updateSortSelection: PropTypes.func, 
}

export default SortBar;

