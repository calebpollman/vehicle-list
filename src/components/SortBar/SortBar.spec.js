import React from 'react';
import {shallow} from 'enzyme';
import SortBar from './SortBar';

describe('SortBar', () => {
  let wrapper;
  const sortOptions = ['Year', 'Make', 'Model'];

  beforeEach(() => {
    wrapper = shallow(
      <SortBar 
        sortSelection="Year"
        sortOptions={sortOptions}
      />
    );
  });

  it('should render a wrapper <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render an instance of <SortButton /> for each index of sortOptions', () => {
    expect(wrapper.find('SortButton').length).toEqual(sortOptions.length);
  });
});