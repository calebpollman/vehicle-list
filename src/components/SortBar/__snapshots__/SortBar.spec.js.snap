// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SortBar
    sortOptions={
        Array [
            "Year",
            "Make",
            "Model",
          ]
    }
    sortSelection="Year"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="sort-bar-text"
>
          Sort Vehicles By:
</label>,
        Array [
          <SortButton
            isActive={false}
            sortOption="Year"
            updateSortSelection={undefined}
/>,
          <SortButton
            isActive={false}
            sortOption="Make"
            updateSortSelection={undefined}
/>,
          <SortButton
            isActive={false}
            sortOption="Model"
            updateSortSelection={undefined}
/>,
        ],
      ],
      "className": "sort-bar-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sort Vehicles By:",
          "className": "sort-bar-text",
        },
        "ref": null,
        "rendered": "Sort Vehicles By:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "Year",
        "nodeType": "function",
        "props": Object {
          "isActive": false,
          "sortOption": "Year",
          "updateSortSelection": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Make",
        "nodeType": "function",
        "props": Object {
          "isActive": false,
          "sortOption": "Make",
          "updateSortSelection": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Model",
        "nodeType": "function",
        "props": Object {
          "isActive": false,
          "sortOption": "Model",
          "updateSortSelection": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="sort-bar-text"
>
            Sort Vehicles By:
</label>,
          Array [
            <SortButton
              isActive={false}
              sortOption="Year"
              updateSortSelection={undefined}
/>,
            <SortButton
              isActive={false}
              sortOption="Make"
              updateSortSelection={undefined}
/>,
            <SortButton
              isActive={false}
              sortOption="Model"
              updateSortSelection={undefined}
/>,
          ],
        ],
        "className": "sort-bar-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sort Vehicles By:",
            "className": "sort-bar-text",
          },
          "ref": null,
          "rendered": "Sort Vehicles By:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "Year",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "sortOption": "Year",
            "updateSortSelection": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Make",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "sortOption": "Make",
            "updateSortSelection": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Model",
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "sortOption": "Model",
            "updateSortSelection": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
